Step 1: Import modules

In [1]:
from sklearn.feature_extraction.text import TfidfVectorizer

Step 2: Collect strings from documents and create a corpus

In [2]:
d0 = 'Geeks for geeks'
d1 = 'Geeks'
d2 = 'r2j'
string = [d0, d1, d2]

Step 3: Get TF-IDF values

Here we are using TfidfVectorizer() from scikit learn to perform tf-idf and apply on our courpus using fit_transform.

In [3]:
tfidf = TfidfVectorizer()
result = tfidf.fit_transform(string)

Step 4: Display IDF values

In [4]:
print('\nidf values:')
for ele1, ele2 in zip(tfidf.get_feature_names_out(), tfidf.idf_):
    print(ele1, ":", ele2)


idf values:
for : 1.6931471805599454
geeks : 1.2876820724517808
r2j : 1.6931471805599454


Step 5: Display TF-IDF values along with indexing

In [5]:
print('\nWord indexes:')
print(tfidf.vocabulary_)
print('\ntf-idf values:')
print(result)
print("\ntf-idf values in matrix form:")
print(result.toarray())


Word indexes:
{'geeks': 1, 'for': 0, 'r2j': 2}

tf-idf values:
<Compressed Sparse Row sparse matrix of dtype 'float64'
	with 4 stored elements and shape (3, 3)>
  Coords	Values
  (0, 1)	0.8355915419449176
  (0, 0)	0.5493512310263033
  (1, 1)	1.0
  (2, 2)	1.0

tf-idf values in matrix form:
[[0.54935123 0.83559154 0.        ]
 [0.         1.         0.        ]
 [0.         0.         1.        ]]


| Document | Word  | Document Index | Word Index | TF-IDF Value |
|-----------|--------|----------------|-------------|---------------|
| d0 | for | 0 | 0 | 0.549 |
| d0 | geeks | 0 | 1 | 0.8355 |
| d1 | geeks | 1 | 1 | 1.000 |
| d2 | r2j | 2 | 2 | 1.000 |
